tests/test_dataset.py

```diff
@@ -2050,11 +2050,16 @@ def test_compression() -> None:
 
     # writing unaligned data to an uncompressed dataset
     write_data = (np.random.rand(3, 10, 20, 30) * 255).astype(np.uint8)
-    mag1.write(write_data)
+
+    # Writing compressed data directly to "mag1" also works, but using a View here covers an additional edge case
```

Comment:
Does this code "write compressed data directly"? Since `compress()` happens after the `write`, I would have thought that this is not the case.. If I'm wrong, please correct me, but otherwise, I would prefer if the test case was extended so that the compressed-write is also covered.

Reply:
Good catch, I adjusted the wrong test cases. 

I updated the `Changelog` in a way that we can create a new release `0.6.7` once this is merged.